### PR TITLE
Restore backend state when Playwright test finishes

### LINKED ISSUE #1094: [Playwright Test] Restore backend state when test finishes
The playwright test currently has sideeffect of causing backend state change such as removing files under `users/workflows` and modify `user.setting.json` setting config file. The model downloading endpoint test can also add/remove files from `models/` directory.

This can be annoying especially when many of our developers don't have a separate comfyui backend setup for running test.

### PR DESCRIPTION
**Pull request by christian-byrne**
Closes https://github.com/Comfy-Org/ComfyUI_frontend/issues/1094

Backup and restore `user` and `models` folders in global setup/teardown of local Playwright tests. 

Backup `user` by making a copy. Backup `models` by renaming and replacing with scaffolding copy (folders only).

Get base ComfyUI path from `DEPLOY_COMFYUI_DIR` var.

### CODE REVIEW

File: browser_tests/globalSetup.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { FullConfig } from '@playwright/test'
+import { backupPath } from './utils/backupUtils'
+import dotenv from 'dotenv'
+
+dotenv.config()
+
+export default function globalSetup(config: FullConfig) {
+  if (!process.env.CI) {
+    if (process.env.DEPLOY_COMFYUI_DIR) {
+      backupPath([process.env.DEPLOY_COMFYUI_DIR, '..', 'user'])
```
Comment (huchenlei) [inline browser_tests/globalSetup.ts]: Let's maybe define another env var as the deployment dir might not always be Comfy/web.

**Review by huchenlei — CHANGES_REQUESTED** [review]
I meet a weild bug that If I cleanup the workflows directory and run `Can show top level saved workflows` test, the generated `workflow1.json` and `workflow2.json` does not get removed. Can you try reproduce that?

Steps:
- Prepare an empty `users/workflows/` directory
- Run `Can show top level saved workflows` under `menu.spec.ts` alone

**Comment by christian-byrne** [comment]
> I meet a weild bug that If I cleanup the workflows directory and run `Can show top level saved workflows` test, the generated `workflow1.json` and `workflow2.json` does not get removed. Can you try reproduce that?
> 
> Steps:
> 
> * Prepare an empty `users/workflows/` directory
> * Run `Can show top level saved workflows` under `menu.spec.ts` alone

Having trouble reproducing. I tried with and without `--ui`, running only that test, running only that test file, etc. What command are you starting with?

**Comment by christian-byrne** [comment]
> I meet a weild bug that If I cleanup the workflows directory and run `Can show top level saved workflows` test, the generated `workflow1.json` and `workflow2.json` does not get removed. Can you try reproduce that?
> 
> Steps:
> 
> * Prepare an empty `users/workflows/` directory
> * Run `Can show top level saved workflows` under `menu.spec.ts` alone

Having trouble reproducing. I tried with and without `--ui`, running only that test, only that test file, etc. What command are you launching with?

File: browser_tests/globalSetup.ts (hunk 1 above)
Comment (christian-byrne) [reply browser_tests/globalSetup.ts]: Changed to `TEST_COMFYUI_DIR`

**Review by huchenlei — APPROVED** [review]

**Pull request merged**